lib/client/backend_client.dart

```diff
@@ -23,8 +22,25 @@ class BackendClient {
     }
   }
 
-  void postData(String data) {
-    //add task to API @PostRequest
+  Future<dynamic> postData({required String uri, dynamic body}) async {
+    try {
+      var url = Uri.http(localhost, uri);
```

Comment:
This is a good example of an opportunity to create a `final` variable. `final` variables are more performant than regular local variables. `const` variables are the most performant because they can be set at compile-time. This can't be a `const`, but we can set it to `final`.

I generally put the type (in this case `Uri`) because it helps my eyes, but that's a preference thing as long as Flutter still knows the type (i.e. it shows if you hover over the variable name). Both `final url` or `final Uri url` work well.

Untested suggestion:
```suggestion
      final Uri url = Uri.http(localhost, uri);
```

Reply:
done